screens/ServerScreen.tsx

```diff
@@ -12,63 +12,91 @@ import { useTranslation } from "react-i18next";
 import { NativeStackScreenProps } from "@react-navigation/native-stack";
 import { RootStackParamList } from "types";
 
-export interface Entry {
-  title: string;
-  url: string;
-}
-
 export default function ServerScreen({
   navigation,
 }: NativeStackScreenProps<RootStackParamList, "Server">) {
   const { t } = useTranslation();
   const [searching, setSearching] = useState(false);
   const [finished, setFinished] = useState(false);
   const [scanNotPossible, setScanNotPossible] = useState(false);
-  const [found, setFound] = useState<Entry[]>([]);
+  const [found, setFound] = useState<Set<ServiceDiscovery.Service>>(new Set());
+
   const { updateServerUrl } = useAppContext();
 
   const scanNetwork = useCallback(() => {
+    // for multiple clicks on button
+    ServiceDiscovery.stopSearch("http");
+    ServiceDiscovery.stopSearch("https");
+
     setSearching(true);
     setFinished(false);
-    setFound([]);
+    setFound(new Set());
 
-    let zeroconf: Zeroconf | null = null;
+    const foundListener = ServiceDiscovery.addEventListener(
+      "serviceFound",
+      (service) => {
+        if (service.name === "evcc") {
+          console.log("Found service ", service);
+          setFound((found) => {
+            let serviceAlreadyInSet = false;
 
-    try {
-      if (zeroconf) {
-        (zeroconf as Zeroconf).stop();
-      }
-      zeroconf = new Zeroconf();
-      zeroconf.scan("http", "tcp", "local.");
-    } catch (e) {
-      console.log("error", e);
-      setSearching(false);
-      setScanNotPossible(true);
-    }
-
-    zeroconf?.on("resolved", ({ txt, name, host, port }) => {
-      console.log("resolved", name);
-      if (txt && name?.includes("evcc")) {
-        console.log("found evcc", name);
-        // remove trailing dots
-        const entry = {
-          title: name,
-          url: `http://${host.replace(/\.$/, "")}${port === 80 ? "" : `:${port}`}${txt["path"]}`,
-        };
-        setFound((prevFound) => [...prevFound, entry]);
+            found.forEach((item) => {
```

Comment:
`found.some((i) => ...)` 

Da sparst du dir it manuelle Variable um einen Treffer zu tracken.